qt/python/mantidqtinterfaces/mantidqtinterfaces/Muon/GUI/Common/phase_table_widget/phase_table_presenter.py

```diff
@@ -89,13 +77,13 @@ def update_current_run_list(self):
             self.view.setEnabled(True)
 
     def update_current_groups_list(self):
-        self.view.set_group_combo_boxes(self.context.group_pair_context.group_names)
+        self.view.set_group_combo_boxes(self.model.group_pair_names)
         self.update_model_from_view()
 
     def cancel_current_alg(self):
         """Cancels the current algorithm if executing"""
-        if self.current_alg is not None:
-            self.current_alg.cancel()
+        if self.model.current_alg is not None:
```

Comment:
Just a thought, but looking at where `self.model.current_alg` is used, instead of exposing properties for `current_alg` could the model potentially just provide a `cancel_current_alg()` method that does all of this, plus a `clear_current_alg()` method that simply sets `self._current_alg = None`. Maybe that way we encapsulate just a bit more of the management of `current_alg` in the model?